catalog/project/kcc-namespace/kcc-project-owner.yaml

```diff
@@ -0,0 +1,29 @@
+# Copyright 2021 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# Grant KCC owner of this project
+apiVersion: iam.cnrm.cloud.google.com/v1beta1
+kind: IAMPolicyMember
+metadata:
+  name: kcc-project-id-owners-permissions # {"$kpt-set":"project-id-sa-owners-permissions"}
+  namespace: projects # {"$kpt-set":"projects-namespace"}
+  annotations:
+    cnrm.cloud.google.com/project-id: management-project-id # {"$kpt-set":"management-project-id"}
+spec:
+  member: "serviceAccount:kcc-project-id@management-project-id.iam.gserviceaccount.com" # {"$kpt-set":"project-id-sa-ref"}
+  resourceRef:
+    apiVersion: resourcemanager.cnrm.cloud.google.com/v1beta1
+    kind: Project
+    external: project-id # {"$kpt-set":"project-id"}
```

Comment:
Are we switching to an external ref?
In that case do we still need `cnrm-viewer` for project ns?
https://github.com/GoogleCloudPlatform/blueprints/blob/2631a66b0892ecff8e9a29b75b89d4b2aae8b243/catalog/project/kcc-namespace/kcc-namespace-viewer.yaml#L36

Reply:
I'm not sure we should switch to external ref. We already have a setter for projects namespace and I think we should continue using that.

if the project is managed outside KCC, we can provide instructions in the tutorial to override this.

Reply:
If I make a tutorial for this and link to it from the KCC guide that describes the manual steps, I'm pretty sure that use case will become the predominant driver of usage, rather than just LZ users. Do we really want to use a default that prioritizes LZ users over the simpler use case?

Reply:
It's not necessarily about prioritizing landing zone users, but rather prioritizing users who are *also* managing the Project in KCC as well. I think it's easier for end users to flip from a reference to `external` (if they are using an externally-managed project) than it is for them to build out the right references.

We should not break landing zone with this change.

Reply:
Switching to external won't break LZ.

> I think it's easier for end users to flip from a reference to external

This requires a manual code edit (or yq use), something I've been trying to avoid because it's harder to automate and requires more complex user instructions in tutorials (can't just copy/paste a terminal command).

If there were a way to do this with a copy/pastable one-liner, I would feel better about requiring it for significant use cases.

Reply:
> If there were a way to do this with a copy/pastable one-liner, I would feel better about requiring it for significant use cases.

Could you use a kpt function (maybe search & replace) to do this? If no existing function allows it, I think `externalize-ref` would be a good general purpose function.

Reply:
The search-replace function doesn't seem to have any way to identify a single resource to modify.

I filed a FR: https://github.com/GoogleContainerTools/kpt/issues/2384

Reply:
Thanks for opening the FR.

Given the current constraints, I'm comfortable with this as-is. I also think we could develop a function to automatically internalize references (see b/193349853).